### ISSUE TITLE
[BUG] PCA not created when n_cells = 2

### ISSUE THREAD

**Issue by jashapiro** [AlexsLemonade/scpca-nf#798]
It seems that the bug below is an uncaught case where there are exactly 2 cells (after processing). In that case, we can't calculate highly variable genes, so we skip PCA entirely. 

```
Error executing process > 'sce_to_anndata:export_anndata (SCPCL001160)'

Caused by:
  Essential container in task exited


Command executed:

  sce_to_anndata.R         --input_sce_file SCPCL001160_processed.rds         --output_rna_h5 SCPCL001160_processed_rna.h5ad         --output_feature_h5 SCPCL001160_processed_null.h5ad         --output_pca_tsv SCPCL001160_processed_pca.tsv                  
  
  # move any normalized counts to X in AnnData, convert matrices, and add PCA metadata
  if [ "processed" = "processed" ]; then
    reformat_anndata.py --anndata_file SCPCL001160_processed_rna.h5ad --pca_meta_file SCPCL001160_processed_pca.tsv
    # move counts in feature data, if the file exists
    if [ -f "SCPCL001160_processed_null.h5ad" ]; then
      reformat_anndata.py --anndata_file SCPCL001160_processed_null.h5ad --hvg_name "none"
    fi
  fi

Command exit status:
  1

Command output:
  (empty)

Command error:
  sce read
  Formatting done
  Registered S3 method overwritten by 'zellkonverter':
    method                                             from      
    py_to_r.pandas.core.arrays.categorical.Categorical reticulate
  singler_results cannot be converted between SCE and AnnData.
  /usr/local/renv/basilisk/1.16.0/zellkonverter/1.14.1/zellkonverterAnnDataEnv-0.10.6/lib/python3.12/site-packages/anndata/_core/anndata.py:430: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.
    warnings.warn(
  Exported RNA
  Traceback (most recent call last):
    File "//nextflow-bin/reformat_anndata.py", line 70, in <module>
      raise FileExistsError("`pca_meta_file` does not exist.")
  FileExistsError: `pca_meta_file` does not exist.

Work dir:
  s3://nextflow-ccdl-data/work/8e/e3267ca5a43b81f8c32ee139d0d8b5

Container:
  ghcr.io/alexslemonade/scpcatools-anndata:v0.4.1

Tip: when you have fixed the problem you can continue the execution adding the option `-resume` to the run command line
```

I will investigate this in order to complete https://github.com/AlexsLemonade/ScPCA-admin/issues/986

_Originally posted by @jashapiro in https://github.com/AlexsLemonade/scpca-nf/issues/789#issuecomment-2454862215_

**Comment by jashapiro** [comment]
There are a few possible solutions here: 
1. Skip PCA in this case completely, as we do now, but account for this in later steps that assume a PCA will be present. (I have implemented this already).
2.  Force PCA for 2 cells without calculating highly variable genes (and then account for potentially missing highly variable genes variables) (modify this code: https://github.com/AlexsLemonade/scpca-nf/blob/aea4c80d34930002070481116ec92e9dd7ee79b8/bin/post_process_sce.R#L245-L263)
3.  Change the filter step (https://github.com/AlexsLemonade/scpca-nf/blob/aea4c80d34930002070481116ec92e9dd7ee79b8/modules/export-anndata.nf#L70-L73) to only create anndata files for files with at least 3 cells.

All of these are pretty easy, but I think I favor option 3, and then maybe designating this sample as low quality so we don't include it on the portal?

A note that this seems to have appeared with some modification (or change in seed) for `miQC` where a sample that previously had 7 cells after filtering there now has only 2.

**Comment by jaclyn-taroni** [comment]
Option 3 would work for me.

**Comment by jashapiro** [comment]
closed by #799
